3.x Fix console output configuration error

davprat opened the pull request:
### Description of changes
* Removed support for updating compute console logging property in clustermgtd config.
  * Value was based off of whether or not CloudWatch  was enabled which is not an updatable setting.

### Tests
* Ran test_update.py::test_update_slurm: test.
  * Update recipe successful (test failed due to  InvalidNetworkInterface.InUse)

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

davprat committed:
Fix console output configuration error on update

Signed-off-by: David Pratt <davprat@amazon.com>

jdeamicis commented on cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb:
```diff
@@ -176,7 +176,7 @@ def update_nodes_in_queue(strategy, queues)
   path "/etc/parallelcluster/slurm_plugin/parallelcluster_clustermgtd.conf"
   pattern "compute_console_logging_enabled*"
-  line "compute_console_logging_enabled = #{node['cluster']['config'].dig(:Monitoring, :Logs, :CloudWatch, :Enabled) || node['cluster']['slurm_console_logging']['enabled']}"
+  line "compute_console_logging_enabled = <%= node['cluster']['cw_logging_enabled'] == 'true' && node['cluster']['slurm_console_logging']['enabled'] %>"
```
For better readability and consistency with the rest of the code, I believe we should use a `lazy {}` construct with a `not_if` condition on the whole `replace_or_add` block. Something similar to what we do [here](https://github.com/aws/aws-parallelcluster-cookbook/blob/5377d823e239ebb6537d68f907ea699fdb97618c/cookbooks/aws-parallelcluster-config/recipes/update_shared_storages.rb#L203-L211).

What do you think?

davprat replied:
I've removed the `['cluster']['slurm_console_logging']['enabled']` altogether since it wasn't really serving any useful purpose and just simplified to use the value provided by `['cluster']['cw_logging_enabled']`.

davprat replied:
I've removed this entire resource in the last commit.

davprat committed:
Remove unnecessary node property

This change simplifies the logic for enabling compute console
output logging to be based soley on if CloudWatch logging is
enabled.

Signed-off-by: David Pratt <davprat@amazon.com>

chenwany commented on cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb:
```diff
@@ -175,9 +175,8 @@ def update_nodes_in_queue(strategy, queues)
-  line "compute_console_logging_enabled = #{node['cluster']['config'].dig(:Monitoring, :Logs, :CloudWatch, :Enabled) || node['cluster']['slurm_console_logging']['enabled']}"
-  replace_only true
+  pattern "compute_console_logging_enabled.*"
+  line "compute_console_logging_enabled = #{node['cluster']['cw_logging_enabled']}"
```
Currently, `cw_logging_enabled` setting doesn't support update during cluster update, so I think we can remove the code here

davprat replied:
Removed.

davprat committed:
Remove update of console output logging config

Signed-off-by: David Pratt <davprat@amazon.com>

chenwany reviewed (approved):
LGTM!
